runtime/moonriver/src/lib.rs

```diff
@@ -1022,6 +1023,22 @@ impl pallet_maintenance_mode::PauseXcmExecution for XcmExecutionManager {
 	}
 }
 
+pub struct NormalDmpHandler;
+impl DmpMessageHandler for NormalDmpHandler {
+	// This implementation makes messages be queued
+	// Since the limit is 0, messages are queued for next iteration
+	fn handle_dmp_messages(
```

Comment:
Does this imply a new db read every block where we have dmp messages? I guess its similar to what happens in xcmp-queue, but I wonder whether we can account fo this somehow

Reply:
Yes it imply 1 db read at each block. Maybe the best place to account it is the BaseBlockWeight ?

Reply:
Might well be, yes

Reply:
CC: @notlesh

Reply:
Although this read only occurs if we have dmp messages. I think we can maybe add the additional weight to the return of such function?

Reply:
Basing my thought on https://github.com/paritytech/cumulus/blob/b566f3b3038dd1af0667613e4477fb1fd4478cb3/pallets/parachain-system/src/lib.rs#L824

Reply:
I finally added the additional read to the returned weight of `handle_dmp_messages`